### PR TITLE
Upgrade travis setup

### PR DESCRIPTION
**Pull request by chenrui333**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,11 @@
-sudo: required
```
Comment (chenrui333) [inline .travis.yml]: this has been deprecated, https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

**Comment by whummer** [comment]
Thanks for tackling this migration @chenrui333 ! Looks like there's still a minor issue with the `oracle-java8-installer` package not being found.

**Comment by chenrui333** [comment]
yeah, that is right. I will check how to fix that later.

**Comment by DavidTanner** [comment]
I think you will need to switch to openjdk-8-jre-headless

**Comment by whummer** [comment]
> I think you will need to switch to openjdk-8-jre-headless

Thanks @DavidTanner . @chenrui333 can we give that a try?

**Comment by chenrui333** [comment]
Sure, definitely.

**Comment by chenrui333** [comment]
```
======================================================================
ERROR: Failure: ModuleNotFoundError (No module named 'google_compute_engine')
----------------------------------------------------------------------
```

**Comment by chenrui333** [comment]
```
[success] 0.01% tests.unit.test_sns_listener.SNSTests.test_create_sns_message_body_raw_message_delivery: 0.0002s
[error] 0.00% <nose.suite.ContextSuite context=tests.integration>:setup: 0.0000s
----------------------------------------------------------------------
```

looks like integration test suite failed now.

**Comment by chenrui333** [comment]
@whummer any idea how to troubleshoot `tests.integration` failure?

**Comment by chenrui333** [comment]
Refresh the branch with latest master.

**Comment by chenrui333** [comment]
Still failing with the `tests.integration` part.

**Comment by DavidTanner** [comment]
I'm seeing that as an intermittent failure on my branch.  It doesn't always fail.

**Comment by chenrui333** [comment]
So what should be the right way to handle this?

**Comment by DavidTanner** [comment]
It seems to be always an issue with `start_sns` in `infra.py`

**Comment by chenrui333** [comment]
I see, looks like the build is green now. I will squash some of my changes.

**Comment by chenrui333** [comment]
@whummer, should be good for merge now.

**Comment by whummer** [comment]
Great, thanks for digging into this @chenrui333 !

**Pull request merged**